jenkins/check_projects.py

```diff
@@ -39,190 +41,393 @@
 import subprocess
 import sys
 import threading
+import logging
 
-semaphore = threading.Semaphore(10)  # 10 concurrent gcloud calls
-lock = threading.Lock()  # Don't print/append concurrently
-null = open('/dev/null', 'w')
-
-# Not thread safe, acquire lock before accessing.
-completed = set()  # Projects with no updated needed
-updated = set()  # Projects which changed
-errors = set()  # Projects which failed to fully update
-helpers = {}  # People with owners rights to update error projects
+logger = logging.getLogger('check_project')
 
 DEFAULT = {
-    'roles/editor': [
+    'IAM': {
+      'roles/editor': [
         'serviceAccount:kubekins@kubernetes-jenkins.iam.gserviceaccount.com',
-        'serviceAccount:pr-kubekins@kubernetes-jenkins-pull.iam.gserviceaccount.com',
-    ],
+        'serviceAccount:pr-kubekins@kubernetes-jenkins-pull.iam.gserviceaccount.com'],
+    },
+    'EnableVMZonalDNS': False,
 }
 
 
+class RateLimitedExec(object):
+  def __init__(self):
```

Comment:
Sadly Sen is right that expected external style is to indent with 4 spaces (*extremely*, gross, I know). Although if you act quick I promise to look the other way :)

https://google.github.io/styleguide/pyguide.html?showone=Indentation#Indentation